src/devices/bus/mononcol/slot.h

```diff
@@ -0,0 +1,143 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+#ifndef MAME_BUS_MONONCOL_SLOT_H
+#define MAME_BUS_MONONCOL_SLOT_H
+
+#pragma once
+
+#include "imagedev/cartrom.h"
+
+#include <cassert>
+
+/***************************************************************************
+ TYPE DEFINITIONS
+ ***************************************************************************/
+
+class device_mononcol_cart_interface : public device_interface
+{
+public:
+
+	// construction/destruction
+	virtual ~device_mononcol_cart_interface();
+
+	// reading and writing
+	virtual void rom_alloc(u32 size, int width, endianness_t end, char const *tag);
+
+	u8 *get_rom_base()  { return m_rom; }
+	u32 get_rom_size() { return m_rom_size; }
+
+	virtual uint8_t read() { return 0x00; }
+	virtual DECLARE_WRITE_LINE_MEMBER(dir_w) { }
+	virtual void write(uint8_t data) { };
+	virtual void set_ready() { };
+
+	virtual void set_spi_region(uint8_t* region) { }
+	virtual void set_spi_size(size_t size) { }
+
+protected:
+	device_mononcol_cart_interface(machine_config const &mconfig, device_t &device);
+
+	// internal state
+	u8 *m_rom;
+	u32 m_rom_size;
+};
+
+
+#define MONONCOL_ROM_REGION_TAG ":cart:rom"
```

Comment:
This definitely shouldn’t be in the header – how the cartridges allocate memory is an implementation detail.  Exposing it downstream encourages leaky abstractions.

Reply:
Yeah, remove this please

Reply:
gone